src/server/scripts/EasternKingdoms/zone_dun_morogh.cpp

```diff
@@ -83,7 +96,639 @@ class npc_frozen_mountaineer : public CreatureScript
+                        break;
+                    }
+                    case 7:
+                    {
+                        agent = me->FindNearestCreature(NPC_FROSTMANE_RAIDER, 25.0f, true);
+                        agent->AI()->Talk(TALK_0);
+                        SummonTimer = 10500;
+                        Phase++;
+                    }
+
+                    case 8:
+                    {
+                        Creature* Troll1 = ObjectAccessor::GetCreature(*me, Troll1GUID);
+                        Creature* Troll2 = ObjectAccessor::GetCreature(*me, Troll2GUID);
+                        Creature* Troll3 = ObjectAccessor::GetCreature(*me, Troll3GUID);
+                        Creature* Troll4 = ObjectAccessor::GetCreature(*me, Troll4GUID);
+                        Troll1->SetReactState(REACT_AGGRESSIVE);
+                        Troll1->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                        Troll2->SetReactState(REACT_AGGRESSIVE);
+                        Troll2->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                        Troll3->SetReactState(REACT_AGGRESSIVE);
+                        Troll3->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                        Troll4->SetReactState(REACT_AGGRESSIVE);
+                        Troll4->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+                        SummonTimer = 11000;
+                        Phase++;
+                        break;
+                    }
+                    case 9:
+                    {
+                        Creature* Troll1 = ObjectAccessor::GetCreature(*me, Troll1GUID);
+                        Creature* Troll2 = ObjectAccessor::GetCreature(*me, Troll2GUID);
+                        Creature* Troll3 = ObjectAccessor::GetCreature(*me, Troll3GUID);
+                        Creature* Troll4 = ObjectAccessor::GetCreature(*me, Troll4GUID);
+                        Troll1->CombatStart(player, true);
+                        Troll2->CombatStart(player, true);
+                        Troll3->CombatStart(player, true);
+                        Troll4->CombatStart(player, true);
+                        SummonTimer = 20000;
+                        Phase++;
+                        break;
+                    }
+                    case 10:
+                    {
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID))
+                            player->KilledMonsterCredit(NPC_AMBERSTILL_RAID_CONTROLLER, PlayerGUID);
+                        SummonTimer = 30000;
+                        break;
+                    }
+
+                    default:
+                        break;
+                    }
+                }
+            }
+            else SummonTimer -= diff;
+        }
+
+    private:
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_sergeant_flinthammerAI(creature);
+    }
+};
```

Comment:
use mby RegistSpellScrept / RegisterNpcScript ? it'll make just 300 lines of code insted of 500